libfastpm/cosmology.c

```diff
@@ -480,6 +503,21 @@ double ComovingDistance(double a, FastPMCosmology * c) {
     return result;
 }
 
+double ComovingDistance(double a, FastPMCosmology * c) {
+    struct FastPMComovingDistanceInterp * CDInterp = c->CDInterp;
+    double y;
+    int status;
+    status = gsl_interp_eval_e(CDInterp->interp, CDInterp->x, CDInterp->y, a, CDInterp->accel, &y);
```

Comment:
Got error "srun: error: nidxxxx: task 0: Segmentation fault; srun: launch/slurm: _step_signal: Terminating StepId=xxxxxxx.x". I use gdb core mode to debug and got following message:
>>Core was generated by `/fastpm/tests/./testsubsample'.
>>Program terminated with signal SIGSEGV, Segmentation fault.
>>#0  gsl_interp_eval_e (interp=0x0, xa=0x7fffffff1ab8, ya=0x7fffffff3ac0, x=0.20000000000000001, a=0x0, y=0x7fffffff1998)
    at interp.c:131
131     interp.c: No such file or directory.

>>(gdb) bt
>>#0  gsl_interp_eval_e (interp=0x0, xa=0x7fffffff1ab8, ya=0x7fffffff3ac0, x=0.20000000000000001, a=0x0, y=0x7fffffff1998)
    at interp.c:131
>>#1  0x000000002000b2da in ComovingDistance (a=a@entry=0.20000000000000001, c=c@entry=0x7fffffff1a20) at cosmology.c:510
>>#2  0x00000000200018de in main (argc=<optimized out>, argv=<optimized out>) at testsubsample.c:64

seems like line 507-510 causes some storage problem.

Reply:
We were missing a fastpm_cosmology_init in the tests.